collectors/freeipmi.plugin/freeipmi_plugin.c

```diff
@@ -1118,486 +883,869 @@ _ipmimonitoring_sensors (struct ipmi_monitoring_ipmi_config *ipmi_config)
-             * event offset type, and the specific event offset that occurred.
-             *
-             * See ipmi_monitoring_offsets.h for a list of event offsets
-             * and types.
-             */
+static int *excluded_status_record_ids = NULL;
+size_t excluded_status_record_ids_length = 0;
 
-            if (!(sensor_name = ipmi_monitoring_sel_read_sensor_name (ctx)))
-            {
-                collector_error( "ipmi_monitoring_sel_read_sensor_name(): %s",
-                                 ipmi_monitoring_ctx_errormsg (ctx));
-                goto cleanup;
-            }
+static void excluded_status_record_ids_parse(const char *s, bool debug) {
+    if(!s) return;
 
-            if ((sensor_type = ipmi_monitoring_sel_read_sensor_type (ctx)) < 0)
-            {
-                collector_error( "ipmi_monitoring_sel_read_sensor_type(): %s",
-                                 ipmi_monitoring_ctx_errormsg (ctx));
-                goto cleanup;
-            }
+    while(*s) {
+        while(*s && !isdigit(*s)) s++;
 
-            if ((sensor_number = ipmi_monitoring_sel_read_sensor_number (ctx)) < 0)
-            {
-                collector_error( "ipmi_monitoring_sel_read_sensor_number(): %s",
-                                 ipmi_monitoring_ctx_errormsg (ctx));
-                goto cleanup;
-            }
+        if(isdigit(*s)) {
+            char *e;
+            unsigned long n = strtoul(s, &e, 10);
+            s = e;
 
-            if ((event_direction = ipmi_monitoring_sel_read_event_direction (ctx)) < 0)
-            {
-                collector_error( "ipmi_monitoring_sel_read_event_direction(): %s",
-                                 ipmi_monitoring_ctx_errormsg (ctx));
-                goto cleanup;
+            if(n != 0) {
+                excluded_status_record_ids = reallocz(excluded_status_record_ids, (excluded_status_record_ids_length + 1) * sizeof(int));
+                excluded_status_record_ids[excluded_status_record_ids_length++] = (int)n;
             }
+        }
+    }
+
+    if(debug) {
+        fprintf(stderr, "%s: excluded status record ids:", program_name);
+        size_t i;
+        for(i = 0; i < excluded_status_record_ids_length; i++) {
+            fprintf(stderr, " %d", excluded_status_record_ids[i]);
+        }
+        fprintf(stderr, "\n");
+    }
+}
+
+
+static int excluded_record_ids_check(int record_id) {
+    size_t i;
+
+    for(i = 0; i < excluded_record_ids_length; i++) {
+        if(excluded_record_ids[i] == record_id)
+            return 1;
+    }
+
+    return 0;
+}
+
+static int excluded_status_record_ids_check(int record_id) {
+    size_t i;
+
+    for(i = 0; i < excluded_status_record_ids_length; i++) {
+        if(excluded_status_record_ids[i] == record_id)
+            return 1;
+    }
+
+    return 0;
+}
+
+static void sensor_set_value(struct sensor *sn, void *sensor_reading, struct netdata_ipmi_state *state) {
```

Comment:
Hello @ktsaou ,

We only have a warning for this function:

 ```sh
collectors/freeipmi.plugin/freeipmi_plugin.c: In function 'sensor_set_value':
collectors/freeipmi.plugin/freeipmi_plugin.c:1409:98: warning: unused parameter 'state' [-Wunused-parameter]
 1409 | static void sensor_set_value(struct sensor *sn, void *sensor_reading, struct netdata_ipmi_state *state) {
      |                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
```

I am going to run coverity on PR right now.

Reply:
This issue was fixed with last commit, and probably coverity report was also fixed, I am running again to confirm.